src/stores/userImage.ts

```diff
@@ -0,0 +1,354 @@
+import { nanoid } from 'nanoid'
+import { defineStore, storeToRefs } from 'pinia'
+import {
+  doc,
+  Timestamp,
+  getDoc,
+  setDoc,
+  updateDoc,
+  DocumentReference,
+  DocumentData,
+} from 'firebase/firestore'
+import {
+  ref,
+  uploadBytesResumable,
+  getDownloadURL,
+  deleteObject,
+} from 'firebase/storage'
+import useAuthStore from '@/stores/auth'
+import { db, storage } from '@/firebase/index'
+
+export interface UploadedImage {
+  userUid: string
+  id: string
+  storageURL: string // for access to storage
+  fileName: string // ex) filename.png
+  fileType: string // ex) image/jpeg
+  fileExtention: string // ex) png
+  createdAt: Timestamp
+  show: boolean // Toolbarに表示・非表示
+  canvases: Record<string, number>
+}
+
+export type UserImageStorage = Record<string, UploadedImage>
+
+const useStoreUserImage = defineStore({
+  id: 'user-image',
+  state: () => ({
+    isLoadingImages: true, // true=読み込み中
+    uploadedImages: [] as UploadedImage[],
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    userImageStorage: {} as Record<string, UploadedImage>,
+  }),
+
+  actions: {
+    // userImageStorageのオブジェクトを取得 Record<string, UploadedImage>
+    async getImageDocSnap(
+      docRef: DocumentReference<DocumentData>,
+    ): Promise<UserImageStorage | undefined> {
+      // firestore上で削除
+      const docSnap = await getDoc(docRef)
+      if (docSnap.exists()) {
+        const userImages: UserImageStorage = docSnap.data().images
+        if (userImages !== undefined) return userImages
+      }
+      return undefined
+    },
+
+    // Toolbarのリストに画像を追加する
+    addImageList(e: Event) {
+      const inputElement = e.target as HTMLInputElement
+      if (inputElement === null) return
+      const { files } = inputElement
+      if (files === null) return
+      const file = files.item(0)
+      if (file === null) return
+      this.uploadImageToStorage(file)
+    },
+
+    // Storageに画像をアップロードする。
+    uploadImageToStorage(file: File) {
+      const { uid } = storeToRefs(useAuthStore())
```

Comment:
7回登場しているので、32行目 exportの前に宣言して良いと思います

Reply:
Storeの外で他のStoreのstateを宣言すると、以下のエラーが出ちゃいます。
`Uncaught ReferenceError: Cannot access 'useAuthStore' before initialization `

応急処置的ですが、uidを取得するメソッドをuserImage.tsに作って呼び出すようにしました。
```suggestion
      getUid() {
        const { uid } = storeToRefs(useAuthStore())
        return uid
      },
      uploadImageToStorage(file: File) {
        // const { uid } = storeToRefs(useAuthStore())
        const uid = this.getUid()
```

Reply:
僕も試しましたができないですね。申し訳ないです
idをgetするだけなら関数化しなくても良いのかなと個人的に思います。